webpack/src/Extends/Host/MonitoringTab/index.js

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListGroup,
+  DescriptionListDescription,
+  Grid,
+  GridItem,
+} from '@patternfly/react-core';
+import { Table, Thead, Tbody, Tr, Th, Td } from '@patternfly/react-table';
+import { STATUS } from 'foremanReact/constants';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { get } from 'foremanReact/redux/API';
+import { selectAPIStatus, selectAPIResponse } from 'foremanReact/redux/API/APISelectors';
+import Loading from 'foremanReact/components/Loading';
+import SkeletonLoader from 'foremanReact/components/common/SkeletonLoader';
+import DefaultLoaderEmptyState from 'foremanReact/components/HostDetails/DetailsCard/DefaultLoaderEmptyState';
+import CardTemplate from 'foremanReact/components/HostDetails/Templates/CardItem/CardTemplate';
+
+const STATUS_ICONS = {
+  'ok': 'pficon-ok status-ok',
+  'warning': 'pficon-info status-warn',
+  'critical': 'pficon-error-circle-o status-error',
+}
+
+const status_icon_class = (result_status) => {
+  return STATUS_ICONS[result_status] || 'pficon-help status-question';
+};
+
+const MonitoringResults = ({ hostId }) => {
+  const dispatch = useDispatch();
+  const API_KEY = `get-monitoring-results-{hostId}`;
+  const status = useSelector(state => selectAPIStatus(state, API_KEY));
+  const { results, itemCount, response } = useSelector(state =>
+    selectAPIResponse(state, API_KEY)
+  );
+
+  const fetchResults = useCallback(
+    () => {
+      if (!hostId) return;
+      dispatch(
+        get({
+          key: API_KEY,
+          url: `/api/hosts/${hostId}/monitoring/results`,
+          params: {
+            per_page: 'all',
+          },
+        })
+      );
+    },
+    [API_KEY, dispatch, hostId],
+  );
+
+  useEffect(() => {
+    fetchResults();
+  }, [fetchResults]);
```

Comment:
Can you use [`useAPI`](https://github.com/theforeman/foreman/blob/a1142ab9ef500e3206be1ff8a390431bc1b32de3/webpack/assets/javascripts/react_app/common/hooks/API/APIHooks.js#L18) instead?

e.g.:

```js
const {
    response: { results },
    status,
  } = useAPI('get', `/api/hosts/${hostId}/monitoring/results`, `get-monitoring-results-{hostId}`);
```

Reply:
I copied it from https://github.com/theforeman/foreman_monitoring/pull/94 so I wonder why that doesn't use `useAPI`.

Reply:
After playing with this for a while I think `useAPI` is an awkward API if you want to handle the error properly (like display Permission Denied). The `response` object is suddenly an instance of `Error` (from axios) that has a property `response`, which is not obvious.

Speaking of error handling, I wish there was some generic way to show an error.

Reply:
we are indeed not very consistent across the app,
though basically `useAPI` came to make things simpler, you can see it's less lines, imports and complexity..
it's implementation is basically what you did here which is also fine.

you can pass to the `useAPI` some options including:
- [`errorToast`](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/redux/API/APIRequest.js#L82): callback function that will trigger a toast notification with a message that you can set based on the error. e.g: `errorToast: error => error`
- [`handleError`](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/redux/API/APIRequest.js#L85C5-L85C16): also a callback that you can choose what to do with.
- same also for `successToast` and `handleSuccess`

Hope it helps